src/libostree/ostree-repo.c

```diff
@@ -2771,18 +2784,22 @@ reload_core_config (OstreeRepo          *self,
       self->zlib_compression_level = OSTREE_ARCHIVE_DEFAULT_COMPRESSION_LEVEL;
   }
 
-  { g_autofree char *min_free_space_percent_str = NULL;
+  { g_autofree char *min_free_space_str = NULL;
+    self->min_free_space_str = NULL;
+
     /* If changing this, be sure to change the man page too */
-    const char *default_min_free_space = "3";
+    const char *default_min_free_space = "3%";
 
+    /* TODO : change config name and update docs */
     if (!ot_keyfile_get_value_with_default (self->config, "core", "min-free-space-percent",
                                             default_min_free_space,
-                                            &min_free_space_percent_str, error))
+                                            &min_free_space_str, error))
       return FALSE;
 
-    self->min_free_space_percent = g_ascii_strtoull (min_free_space_percent_str, NULL, 10);
-    if (self->min_free_space_percent > 99)
-      return glnx_throw (error, "Invalid min-free-space-percent '%s'", min_free_space_percent_str);
+   if (min_free_space_str_validate (min_free_space_str))
+     self->min_free_space_str = g_strdup (min_free_space_str);
```

Comment:
`self->min_free_space_str = g_steal_pointer (&min_free_space_str);` and avoid an extra strdup.